rfcs/1-declarative-configuration.md

```diff
@@ -0,0 +1,127 @@
+# Declarative Configuration
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation Details](#implementation-details)
+  - [Configuration example](#examples)
+- [Test Plan](#testrelease-plan)
+<!-- /toc -->
+
+
+## Summary
+
+This proposal replaces environment variable based configuration being used in Make+Magefiles in favor of using declarative yaml based configuration.
+
+## Motiviation
+
+As we've been writing `devbase`, and other tooling, we've identified that trying to codify all options for tooling to environment variables, bespoke configuration files, and other obscure means has made it difficult for us to provide options and maintain a common set of toolign.
+
+This, combined with a lack of documentation into all of these options, has laregely made `devbase` unconsumable outside of bootstrap/stencil. Thinking of `devbase` as it's own framework to build software makes it more usable in non-stencil projects (like we do already today w/ submodules)
+### Goals
+
+ - Documentation for all options in configuration file(s)
```

Comment:
I keep having an issue with how vague the word options is

Reply:
It's somewhat intentionally vague as it's different dependent on the target, I could all it configuration instead? However that seems equally vague.

Reply:
Maybe this should be comments for all options in config files. I think keeping this info as close to the config as possible will have a big benefit.

Reply:
PTAL, I updated this to be more clear about what I'm envisioning

Reply:
Big fan of the new direction of levering Go docs.